### PR TITLE
HPCC-27773 - Create confidence tests and key files for the General Neural Network bundle

### PR DESCRIPTION
**Pull request by NSel1727**
- Create ecl and key folder
- Develop test code to create confidence tests calling functions from the General Neural Network bundle
- Create a key file for each test, displaying the passing output in XML format

### CODE REVIEW

**Comment by NSel1727** [comment]
@AttilaVamos @ghalliday Please review my pull request

**Comment by NSel1727** [comment]
The ClassicTestModified file may only pass depending on the versions

Here are my versions (passes that test)

Python: 3.8.1
TensorFlow: 2.9.1
HPCC Platform: 8.6

File: ecl/!SetupTestOBT.ecl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        # Should only be one.
+        nodeId, unused, result = tr
+        try:
+          import tensorflow.compat.v1 as tf # V2.x
+          tf.disable_v2_behavior()
+        except:
+          try:
+            import tensorflow as tf # V 1.x
+          except:
+            return [(nodeId, 'unknown', tb.format_exc())]
```
Comment (AttilaVamos) [inline ecl/!SetupTestOBT.ecl]: I think we know the reason of this exception. The message can be ' Failed to import tensorflow. Is it installed?' or something better to inform us about the problem.

File: ecl/ClassicTestModified.ecl [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                  SELF.sqError := SELF.error * SELF.error), LEFT OUTER, LOCAL);
+
+Result := RECORD
+  REAL err := cmp.error;
+END;
+
+Errors := TABLE(cmp, Result);
+
+// No errors should be above 0.1
+HighErrors := COUNT(Errors(err > 0.1));
```
Comment (AttilaVamos) [inline ecl/ClassicTestModified.ecl]: That's what we are talked yesterday. On my Ubuntu 18.04 + Tensorflow 2.6.2 this check (> 0.1) results > 200 errors.
If I change it to 0.2 then I can get only a couple of errors, and with 0.3 it is passed,
So, the question is the 0.3 limit is acceptable for Tensor flow 2.9.1 and GNN point of view or not?
If the answer is yes, we can use 0.3 for that check. 
If not, then I should ensure it is always executed on >=2.9.1. 
Or we can use stored parameter to input Tensor flow version (the automated test system can query it and pass forward) and set the limit value based on the stored parameter.

File: eclcc.log [hunk 3]
```diff
@@ -0,0 +1,32 @@
+00000000 PRG 2022-06-28 12:52:16.183 11478 11478 "Loading plugin /opt/HPCCSystems/plugins/libunicodelib.so[lib_unicodelib] version = UNICODELIB 1.1.06"
```
Comment (AttilaVamos) [inline eclcc.log]: Please remove this file.

**Review by AttilaVamos — CHANGES_REQUESTED** [review]
Generally and automated testing point of view it seems ok, but there is a question and some requests.

**Comment by NSel1727** [comment]
@ghalliday @RogerDev When available, please review my pull request.

**Comment by lilyclemson** [comment]
@NSel1727   To avoid confusion with GNN bundle's ECL files, please create a top level folder such as ConfTest to include the newly added folders/files. Thanks.  

@RogerDev

**Comment by NSel1727** [comment]
@lilyclemson A new root directory was added for the ecl folder to avoid confusion.

File: OBTTests/ecl/key/ClassicTestModified.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+ <Info><Source>user</Source><Message>Training Status: ModelId = 0, Epoch = 1, LR = 1.0, bs = 1024, Loss = 0.00142032571460249</Message></Info>
```
Comment (AttilaVamos) [inline OBTTests/ecl/key/ClassicTestModified.xml]: Please remove these '<Info>...</Info>' lines from the key file. The Regression Test Engine filters them out.

**Review by AttilaVamos — CHANGES_REQUESTED** [review]

**Comment by lilyclemson** [comment]
It's ready to merge. Please review. Thank you @ghalliday 


CC @RogerDev

**Pull request merged**